src/rust/ide/enso-protocol/src/project_manager.rs

```diff
@@ -0,0 +1,382 @@
+//! Client library for the JSON-RPC-based Project Manager service.
+
+use crate::prelude::*;
+
+use crate::types::UTCDateTime;
+use json_rpc::api::Result;
+use json_rpc::Handler;
+use json_rpc::make_rpc_methods;
+use futures::Stream;
+use serde::Serialize;
+use serde::Deserialize;
+use std::future::Future;
+use uuid::Uuid;
+
+
+
+// =============
+// === Event ===
+// =============
+
+// Project Manager has no notifications, so we create a dummy Notification type for it.
+type Notification = ();
+
+/// Event emitted by the Project Manager `Client`.
+pub type Event = json_rpc::handler::Event<Notification>;
+
+
+
+// ===================
+// === RPC Methods ===
+// ===================
+
+make_rpc_methods! {
+/// An interface containing all the available project management operations.
+trait API {
+    /// Request the project picker to open a specified project. This operation also
+    /// includes spawning an instance of the language server open on the specified project.
+    #[MethodInput=OpenProjectInput,rpc_name="project/open",result=open_project_result,
+    set_result=set_open_project_result]
+    fn open_project(&self, project_id:Uuid) -> OpenProjectResponse;
+
```

Comment:
Im wondering why such wordy style is chosen in this macro. This could easily be converted to something like that (and I think it is much more readable:

```rust
    #[RPC "project/open" OpenProjectInput open_project_result set_open_project_result]
    fn open_project(&self, project_id:Uuid) -> OpenProjectResponse;
```

Reply:
O, really you see this as a more readable?

I like the @notdanilo shape, where you clearly see what is the meaning of tokens, and its looks more like standard Rust code.

Reply:
@wdanilo like @farmaazon said, it's Rust's attributes syntax. I vote to keep it like that, because we can even rework it as procedural macro if it's ever needed.

Reply:
Im ok with it then!